layers/+lang/c-c++/packages.el

```diff
@@ -61,7 +61,10 @@
 
 (defun c-c++/init-clang-format ()
   (use-package clang-format
-    :if c-c++-enable-clang-support))
+    :if c-c++-enable-clang-support
+    :init
+    (when c-c++-enable-clang-format-on-save
+      (add-hook 'c-mode-common-hook 'clang-format/format-on-save))))
```

Comment:
I'd misunderstood `c-mode-common-hook`, as it also seems to run for Java and other C-like languages. Besides, consistency is always good. I'll change it in a minute!